### PR TITLE
Docs for TypeScript support in Wasp

### PR DESCRIPTION
**Pull request by sodic**
These docs cover multiple recently implemented TypeScript features:
 - #589
 - #901
 - #902 
 - #982 
 - #969 
 - #975

### CODE REVIEW

File: web/docs/typescript.md [hunk 1]
```diff
@@ -0,0 +1,264 @@
+---
```
Comment (sodic) [inline web/docs/typescript.md:1]: I experimented on how to best structure this, but putting it all in a single file turned out to make the most sense (at least to me).

Let me know if you have some ideas on how to improve it.

File: web/docs/typescript.md [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+  description: "Some random task",
+  isDone: false,
+}
+
+alert(JSON.stringify(task))
+```
+The mentioned type safety mechanisms also apply here: Changing the task entity in our `.wasp` file changes the imported type, which might throw a type error and warn us that our task definition is outdated.
+
+
+## Backend type support for Operations
```
Comment (infomiho) [inline web/docs/typescript.md:124]:
```suggestion
## Backend type support for Queries and Actions
```

To make it consistent with the next section

File: web/docs/typescript.md [hunk 3]
```diff
@@ -0,0 +1,271 @@
...
+
+  // Thanks to the second type argument in `GetTaskInfo`, the compiler knows the
+  // function must return a value of type `string`.
+  return getInfoMessage(task)
+}
+```
+Everything described above applies to Actions.
+
+## Frontend type support for Queries and Actions
+### Type support for the `useQuery` hook
```
Comment (infomiho) [inline web/docs/typescript.md:173]: I got a little confused with the section. There is a lot of moving parts:
- request payload
- response payload
- error type
- component props

I would maybe introduce these concepts with two examples e.g. one for the request and response, and then add error and component props.

**Review by infomiho — COMMENTED** [review]
Looking good, I've some suggestions on consistency and clarity.

File: web/docs/typescript.md [hunk 4]
```diff
@@ -0,0 +1,271 @@
...
+    throw new HttpError(404)
+  }
+  return getInfoMessage(task)
+}
+```
+To migrate this file to TypeScript, all you have to do is:
+ 1. Change the filename from `queries.js` to `queries.ts`.
+ 2. Write some types.
+
+Let's start by only providing a basic `getInfoMessage` function. We'll see how to properly type the rest of the file in the following sections.
```
[the remark this answers is not in the document]
Comment (sodic) [reply web/docs/typescript.md:56]: Added titles for all files.

File: web/docs/tutorials/todo-app/03-listing-tasks.md [hunk 5]
```diff
@@ -39,6 +39,7 @@ query getTasks {
 }
 ```
 :::caution
+<!-- This block is mostly duplicated in typescript.md -->
```
Comment (infomiho) [inline web/docs/tutorials/todo-app/03-listing-tasks.md:42]: Maybe you can use this to keep it tidy: https://docusaurus.io/docs/markdown-features/react#importing-markdown

File: web/docs/typescript.md (hunk 3 above)
Comment (sodic) [reply web/docs/typescript.md:188]: I dropped the props entirely and separated the rest as suggested.

File: web/docs/tutorials/todo-app/03-listing-tasks.md (hunk 5 above)
Comment (sodic) [reply web/docs/tutorials/todo-app/03-listing-tasks.md:42]: Nice idea, but they're not really identical. I could make them identical, but then I think both pages would suffer content-wise.

**Pull request merged**